src/Confix.Tool/src/Confix.Tool/Variables/Providers/Secret/SecretVariableProvider.cs

```diff
@@ -53,20 +59,50 @@ public ValueTask DisposeAsync()
         return ValueTask.CompletedTask;
     }
 
-    private byte[] Encrypt(ReadOnlySpan<byte> valueToEncrypt, ReadOnlySpan<char> publicKey)
+    private byte[] Encrypt(byte[] valueToEncrypt, string publicKey)
     {
         using RSA rsa = RSA.Create();
         rsa.ImportFromPem(publicKey);
         return rsa.Encrypt(valueToEncrypt, Padding);
     }
 
-    private byte[] Decrypt(ReadOnlySpan<byte> encryptedValue, ReadOnlySpan<char> privateKey)
+    private byte[] Decrypt(ReadOnlySpan<byte> encryptedValue, string privateKey)
     {
         using RSA rsa = RSA.Create();
+        if (_configuration.Password is { } password)
+        {
+            privateKey = DecryptPrivateKey(privateKey, password.ToCharArray());
+        }
+
         rsa.ImportFromPem(privateKey);
+
         return rsa.Decrypt(encryptedValue, Padding);
     }
 
+    private static string DecryptPrivateKey(string encryptedPrivateKey, char[] password)
+    {
+        try
+        {
+            using var textReader = new StringReader(encryptedPrivateKey);
+            var pemReader = new PemReader(textReader, new PasswordFinder(password));
+
+            var keyPair = (AsymmetricCipherKeyPair) pemReader.ReadObject();
+            var privateKey = (RsaPrivateCrtKeyParameters) keyPair.Private;
+
+            var textWriter = new StringWriter();
+            var pemWriter = new PemWriter(textWriter);
+            pemWriter.WriteObject(privateKey);
+
+            return textWriter.ToString();
+        }
+        catch (Exception ex)
+        {
+            throw new ExitException(
+                "Could not decrypt private key. Please check your password.",
+                ex);
+        }
+    }
```

Comment:
why not use this instead?
https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.rsa.importfromencryptedpem?view=net-7.0#system-security-cryptography-rsa-importfromencryptedpem(system-readonlyspan((system-char))-system-readonlyspan((system-byte)))

Reply:
It didnt work with ImportFromEncryptedPem. But i will try again